Dockerfile

```diff
@@ -12,16 +21,33 @@ ARG PROTOBUF_VERSION=4.24.4
 
 ARG BASE_IMAGE=nvcr.io/nvidia/pytorch:24.03-py3
 
-FROM ${BASE_IMAGE}
-
-ARG MAX_JOBS
+FROM ${BASE_IMAGE} AS aligner-bump
 
-# needed in case git complains that it can't detect a valid email, this email is fake but works
-RUN git config --global user.email "worker@nvidia.com"
+ARG ALIGNER_COMMIT
 
 WORKDIR /opt
 
+# NeMo Aligner
+RUN <<"EOF" bash -exu
```

Comment:
sorry, why not keep the old

```
RUN git clone https://github.com/NVIDIA/NeMo-Aligner.git && \
    cd NeMo-Aligner && \
    git pull && \
    if [ ! -z $ALIGNER_COMMIT ]; then \
        git fetch origin $ALIGNER_COMMIT && \
        git checkout FETCH_HEAD; \
    fi && \
    pip install --no-deps -e .
```
what is wrong with that approach?

Reply:
ah i see! this essentially prevents us from overwriting the old aligner

Reply:
yea, exactly